governance/DeprecationMeta.md

```diff
@@ -0,0 +1,27 @@
+# Deprecation Meta
+
+# Overview
+
+This document contains high-level information about O3DE's approach to deprecation.
+
+It details language and terms used when categorizing types of deprecation.
+
+## Philosophy
+
+When modifying existing code that external customers rely on we should strive wherever possible to not break them when making changes.
+
+If code is to be changed or removed, we should let customers know ahead of time with a reasonable notice period (approximately 2 releases (2-3 months))
+
+## Product Deprecation vs API Deprecation
```

Comment:
Should be feature as we are not deprecating all of O3DE :)

Reply:
Good call I can update this wording 👍